aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -73,4 +89,19 @@ private S3Client client() {
     }
     return client;
   }
+
+  @Override
+  public void initialize(Map<String, String> properties) {
+    this.sseType = properties.getOrDefault(AwsCatalogProperties.S3_SSE_TYPE, AwsCatalogProperties.S3_SSE_TYPE_NONE);
```

Comment:
As mentioned in #1767, it would be nice to wrap these properties in AwsCatalogProperties as a utility class and just keep reference to the properties as opposed to enumerating them here an in BaseS3File.